design/gateway-apis-implementation.md

```diff
@@ -97,24 +120,31 @@ It's expected that things that check the status will also check the `observedGen
+the purposes of managing HTTPRoute and TLSRoute objects. Contour is not able to specify the address that the Envoy deployment
+will listen on, and so it will ignore any entries in the Gateway's `spec.addresses` field. For Contour, the key Gateway
+section are the Listeners. These define how an implementation should listen for traffic described by Routes.
 
 For Listeners:
-- Listeners inside Gateways will be merged where possible.
+- Listeners inside Gateways will be merged where possible. Note that only two listeners are supported, HTTP and HTTPS.
 - Listeners that specify a port that the Contour controller considers the default secure port will have a HTTP->HTTPS redirect created for them.
 - Listeners that specify a port that is not exposed via the Envoy service will be rejected.
-- Conflicts within a Gateway will result in the relevant Listeners both being rejected (as there is no way to determine which one was first).
+- Conflicts within a Gateway will result in the both Listeners being rejected (as there is no way to determine which one was first).
 - Listeners are considered mergeable if all the fields out of `hostname`, `port`, and `protocol` match, with some additional rules around TLS.
 - Further merging rules are specified in the detailed design below.
-- Listeners that refer to any other Route than HTTPRoute or TLSRoute will be ignored, and a condition placed on the corresponding `status.listeners[]` object saying that it was ignored because those objects are not supported.
-
+- The listener will be ignored if it refers to a Kind other than `HTTPRoute` or `TLSRoute`. The "ResolvedRefs=False"
+  Listener condition and "InvalidRoutesRef" reason will be surfaced by the gateway controller.
+- The listener will be ignored if it specifies "HTTPS" or "TLS" as the protocol but does not specify a certificate reference.
+  The "ResolvedRefs=False" Listener condition and "InvalidCertificateRef" reason will be surfaced by
+  the gateway controller.
+- Listeners that specify the same port are ignored. The "Conflicted=True" Listener condition and "ProtocolConflict"
+  reason will be surfaced by the gateway controller.
+- A Listener will be ignored if it specifies a protocol of "HTTPS" or "TLS" and mode "Terminate", but does not specify a certificate reference.
+  The "ResolvedRefs=False" Listener condition and "InvalidCertificateRef" reason will be surfaced by the gateway controller.
+- A Listener will be ignored if it specifies an "HTTPS" or "TLS" protocol and a certificate reference that is not a Secret.
+  The "ResolvedRefs=False" Listener condition and "InvalidCertificateRef" reason will be surfaced by the gateway controller.
+- A Listener will be ignored if it specifies "Selector" as a namespace selector type but does not specify matchExpression
+  or matchLabels. The "ResolvedRefs=False" Listener condition and "InvalidRoutesRef" reason will be surfaced by the gateway controller.
```

Comment:
Is this an exhaustive list of all the Conditions + Reasons which should cause Contour to *not* try to program a Listener/its routes? And otherwise, it *should* try to program that listener/routes?

Reply:
The update intends to cover the different Listener condition types and reasons. Let me know if I missed anything.